app/AccountancyModule/PaymentModule/presenters/EducationPresenter.php

```diff
@@ -0,0 +1,160 @@
+        $this->groupFormFactory   = $groupFormFactory;
+    }
+
+    protected function startup() : void
+    {
+        parent::startup();
+        $this->template->setParameters([
+            'unitPairs' => $this->readUnits = $units = $this->unitService->getReadUnits($this->user),
+        ]);
+    }
+
+    public function actionDefault() : void
+    {
+        $this->template->setParameters(['educations' => $this->getEducationsWithoutGroup()]);
+    }
+
+    public function actionNewGroup(int $educationId) : void
+    {
+        $educations = $this->getEducationsWithoutGroup();
+
+        if (! $this->isEditable || ! isset($educations[$educationId])) {
+            $this->flashMessage('Pro tento tábor není možné vytvořit skupinu plateb', 'danger');
+            $this->redirect('default');
+        }
+
+        $this->education = $educations[$educationId];
+        $this->template->setParameters(['education' => $this->education]);
+    }
+
+    /**
+     * @param null $unitId - NEZBYTNÝ PRO FUNKCI VÝBĚRU JINÉ JEDNOTKY
+     */
+    public function actionMassAdd(int $id, ?int $unitId = null) : void
+    {
+        $this->id = $id;
+        $group    = $this->model->getGroup($id);
+
+        if ($group === null || ! $this->isEditable) {
+            $this->flashMessage('Neoprávněný přístup ke skupině.', 'danger');
+            $this->redirect('GroupList:');
+        }
+
+        if ($group->getSkautisId() === null) {
+            $this->flashMessage('Neplatné propojení skupiny plateb se vzdělávací akcí.', 'warning');
+            $this->redirect('Default:');
+        }
+
+        $participants = $this->queryBus->handle(
+            new EducationParticipantListQuery(
+                new SkautisEducationId($group->getSkautisId())
+            )
+        );
+
+        $form = $this['massAddForm'];
+        /** @var MassAddForm $form */
+
+        $personsWithPayment = $this->model->getPersonsWithActivePayment($id);
+
+        $participants = array_filter(
+            $participants,
+            function (Participant $p) use ($personsWithPayment) {
+                return ! in_array($p->getPersonId(), $personsWithPayment, true);
+            }
+        );
+
+        foreach ($participants as $p) {
+            $amount = $p->getPayment();
+            $form->addPerson(
+                $p->getPersonId(),
+                $this->queryBus->handle(new MemberEmailsQuery($p->getPersonId())),
+                $p->getDisplayName(),
+                $amount === 0.0 ? null : $amount
+            );
+        }
+
+        $this->template->setParameters([
+            'group'    => $group,
+            'showForm' => ! empty($participants),
+        ]);
+    }
+
+    protected function createComponentMassAddForm() : MassAddForm
+    {
+        return $this->massAddFormFactory->create($this->id);
+    }
+
+    protected function createComponentNewGroupForm() : GroupForm
+    {
+        Assertion::notNull($this->education);
```

Comment:
Nemůžeme ten presenter rozdělit na dva? Protože momentálně má ten presenter dva vzájemně disjunktní stavy.

Pokud vytáhneme aktuální `default` akci do samostatného presenteru, tak:
- V novém presenteru nebude muset být nullable property `$education`, protože se tam nepoužívá
- V tomto presenteru bude property `$education` vždy non-null.

Pokud chceme seskupovat nějak konceptuálně podobné akce do stejných presenterů, tak můžeme radši pak vytvořit presenter, kde budou default akce z `EducationPresenter`u, `EventPresenter`u a `CampPresenteru`.

Reply:
Nastřelil jsem zde https://github.com/skaut/Skautske-hospodareni/pull/1481

Reply:
Rozdělil jsem to podle tebe na 3 presentery